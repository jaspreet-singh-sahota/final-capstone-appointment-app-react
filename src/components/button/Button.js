import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/button.module.css';

const Button = ({ children }) => (
  <>
    <button className={styles.button} type="submit">{children}</button>
  </>
);

Button.propTypes = {
  children: PropTypes.string,
};

Button.defaultProps = {
  children: '',
};

export default Button;
