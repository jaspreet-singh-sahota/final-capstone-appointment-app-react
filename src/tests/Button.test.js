import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/button/Button';

describe('Button', () => {
  it('should render a <button /> component', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.find('button')).toHaveLength(1);
  });

  it('matches Render snapshot', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});
