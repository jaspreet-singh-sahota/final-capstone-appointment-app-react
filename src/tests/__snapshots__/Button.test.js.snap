// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches Render snapshot 1`] = `ShallowWrapper {}`;
